extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/utils/GeometryUtils.java

```diff
@@ -0,0 +1,515 @@
+			final GeometryCollection multi = (GeometryCollection) geometry;
+			for (int i = 0; i < multi.getNumGeometries(); i++) {
+				final Geometry geo = multi.getGeometryN(i);
+				replacements.addAll(fixRangeOfCoordinates(
+						crs,
+						geo));
+			}
+			return replacements;
+		}
+
+		final Coordinate[] geoCoords = geometry.getCoordinates();
+		final Coordinate modifier = findModifier(
+				crs,
+				geoCoords);
+		replacements.addAll(constructGeometriesOverMapRegions(
+				modifier,
+				geometry));
+		return replacements;
+	}
+
+	/**
+	 * Produce a set of polygons for each region of the map corrected for date
+	 * line and hemisphere crossings. Due to the complexity of going around the
+	 * hemisphere, clip the range.
+	 * 
+	 * Consider a polygon that cross both the hemisphere in the north and the
+	 * date line in the west (-182 92, -182 88, -178 88, -178 92, -182 92). The
+	 * result is two polygons: (-180 90, -180 88, -178 88, -178 90, -180 90)
+	 * (180 90, 180 88, 178 88, 178 90, 180 90)
+	 * 
+	 * @param modifier
+	 * @param geometry
+	 *            - a geometry that may cross date line and/or hemispheres.
+	 * @return
+	 */
+	public static List<Polygon> constructGeometriesOverMapRegions(
+			final Coordinate modifier,
+			final Geometry geometry ) {
+		final Coordinate[] geoCoords = geometry.getCoordinates();
+		final List<Polygon> polygons = new LinkedList<Polygon>();
+		final Geometry world = GeometryUtils.world(
+				geometry.getFactory(),
+				GeoWaveGTDataStore.DEFAULT_CRS);
+
+		// First do the polygon unchanged world
+		final Geometry worldIntersections = world.intersection(geometry);
+		for (int i = 0; i < worldIntersections.getNumGeometries(); i++) {
+			final Polygon polyToAdd = (Polygon) worldIntersections.getGeometryN(i);
+			if (!polygons.contains(polyToAdd)) {
+				polygons.add(polyToAdd);
+			}
+		}
+		// now use the modifier...but just the x axis for longitude
+		// optimization...do not modify if 0
+		if (Math.abs(modifier.x) > 0.0000000001) {
+			final Coordinate[] newCoords = new Coordinate[geoCoords.length];
+			int c = 0;
+			for (final Coordinate geoCoord : geoCoords) {
+				newCoords[c++] = new Coordinate(
+						geoCoord.x + modifier.x,
+						geoCoord.y,
+						geoCoord.z);
+			}
+			final Polygon transposedPoly = geometry.getFactory().createPolygon(
+					newCoords);
+
+			final Geometry adjustedPolyWorldIntersections = world.intersection(transposedPoly);
+			for (int i = 0; i < adjustedPolyWorldIntersections.getNumGeometries(); i++) {
+				final Polygon polyToAdd = (Polygon) adjustedPolyWorldIntersections.getGeometryN(i);
+				if (!polygons.contains(polyToAdd)) {
+					polygons.add(polyToAdd);
+				}
+			}
+		}
+
+		return polygons;
+
+	}
+
+	/**
+	 * Make sure the coordinate falls in the range of provided coordinate
+	 * reference systems's coordinate system.
+	 * 
+	 * @param crs
+	 * @param coord
+	 * @return
+	 */
+	public static Coordinate adjustCoordinateToFitInRange(
```

Comment:
adjustCoordinateDimensionToRange() is just the wrong math to do with y and z - they don't have the same reflection property at the extrema.

90 + 1 degrees latitude is arguably 89 degrees instead of clamping at 90 but its definitely not -89 (you wouldn't want to shift from north to south pole)